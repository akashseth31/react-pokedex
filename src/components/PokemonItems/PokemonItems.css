.items {
    width: 250px;
    height: 250px;
    margin: 10px;
    background-color: #FFF;
    background-repeat: no-repeat;
    border: none;
    border-radius: 5px;
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.16);
    text-transform: capitalize;
    background-size: cover;
    position: relative;
    overflow: hidden;
    cursor: pointer;
    outline: none;
  }

  .short-detail {
    position: absolute;
    text-align: center;
    width: 100%;
    bottom: 0;
    background: rgba(0, 0, 0, 0.5);
    left: 0;
    font-size: 15px;
    color: #fff;
    padding: 10px 10px;
}

.items:hover .short-detail {
  background: rgba(0, 0, 0, 0.7);
}