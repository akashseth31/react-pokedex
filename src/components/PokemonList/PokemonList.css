.pokemon-list {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
    width: 100%;
    padding: 10px;
  }