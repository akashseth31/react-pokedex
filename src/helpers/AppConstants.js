import error404 from './assets/error-404.png';

export const GET_POKEDEX_DATA = 'http://pokeapi.co/api/v2/pokemon/';
export const LIMIT = 100;
export const NOT_FOUND = error404;

