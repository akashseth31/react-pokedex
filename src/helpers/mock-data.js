export const data = [
  {
    "name": "bulbasaur",
    "url": "https://pokeapi.co/api/v2/pokemon/1/",
    "details": {
      "abilities": [
        {
          "ability": {
            "name": "chlorophyll",
            "url": "https://pokeapi.co/api/v2/ability/34/"
          },
          "is_hidden": true,
          "slot": 3
        },
        {
          "ability": {
            "name": "overgrow",
            "url": "https://pokeapi.co/api/v2/ability/65/"
          },
          "is_hidden": false,
          "slot": 1
        }
      ],
      "height": 7,
      "id": 1,
      "sprites": {
        "back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
        "back_female": null,
        "back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/1.png",
        "back_shiny_female": null,
        "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
        "front_female": null,
        "front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/1.png",
        "front_shiny_female": null
      },
      "types": [
        {
          "slot": 2,
          "type": {
            "name": "poison",
            "url": "https://pokeapi.co/api/v2/type/4/"
          }
        },
        {
          "slot": 1,
          "type": {
            "name": "grass",
            "url": "https://pokeapi.co/api/v2/type/12/"
          }
        }
      ],
      "weight": 69
    }
  },
  {
    "name": "ivysaur",
    "url": "https://pokeapi.co/api/v2/pokemon/2/",
    "details": {
      "abilities": [
        {
          "ability": {
            "name": "chlorophyll",
            "url": "https://pokeapi.co/api/v2/ability/34/"
          },
          "is_hidden": true,
          "slot": 3
        },
        {
          "ability": {
            "name": "overgrow",
            "url": "https://pokeapi.co/api/v2/ability/65/"
          },
          "is_hidden": false,
          "slot": 1
        }
      ],
      "height": 10,
      "id": 2,
      "sprites": {
        "back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/2.png",
        "back_female": null,
        "back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/2.png",
        "back_shiny_female": null,
        "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
        "front_female": null,
        "front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/2.png",
        "front_shiny_female": null
      },
      "types": [
        {
          "slot": 2,
          "type": {
            "name": "poison",
            "url": "https://pokeapi.co/api/v2/type/4/"
          }
        },
        {
          "slot": 1,
          "type": {
            "name": "grass",
            "url": "https://pokeapi.co/api/v2/type/12/"
          }
        }
      ],
      "weight": 130
    }
  }
];