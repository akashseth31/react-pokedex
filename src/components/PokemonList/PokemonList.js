import React from 'react';
import './PokemonList.css';

const PokemonList = () => {
  return (
    <section className="pokemon-list">
        List here!
    </section>
  )
}


export default PokemonList;