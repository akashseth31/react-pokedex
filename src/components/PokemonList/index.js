import React from 'react';
import PropTypes from 'prop-types';
import './PokemonList.css';
import PokemonItems from '../PokemonItems';

const PokemonList = ({ data, isFetching, next, showDetails }) => {
  return (
    <section className="pokemon-list">
      {
        data.map(pokemonData => <PokemonItems data={pokemonData} key={pokemonData.name} showDetails={showDetails} />)
      }
      {
        isFetching && <div className="list-footer">Fetching more data...</div>
      }
      {
        !next && <div className="list-footer">That's end of the list!</div>
      }
    </section>
  )
};

PokemonList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    details: PropTypes.shape({
      sprites: PropTypes.shape({
        front_default: PropTypes.string,
      }),
      id: PropTypes.number,
      abilities: PropTypes.arrayOf(PropTypes.any),
      height: PropTypes.number,
      types: PropTypes.arrayOf(PropTypes.any),
    }),
    name: PropTypes.string,
  })).isRequired,
  showDetails: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  next: PropTypes.string,
};

PokemonList.defaultProps = {
  next: null,
};


export default PokemonList;