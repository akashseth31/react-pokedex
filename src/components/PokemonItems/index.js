import React from 'react';
import './PokemonItems.css';
import { NOT_FOUND } from '../../helpers/AppConstants';


const PokemonItems = ({ data, showDetails }) => {
  const {
    name,
    details: { sprites },
  } = data;
  return (
    <button
      onClick={() => showDetails(name)}
      className="items"
      style={{ backgroundImage: `url(${sprites.front_default || NOT_FOUND})` }}
    >
      <div className="short-detail">
        {name}
      </div>
    </button>
  );
};


export default PokemonItems;