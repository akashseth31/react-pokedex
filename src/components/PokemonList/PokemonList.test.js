import React from 'react';
import { mount } from 'enzyme';
import { data } from '../../helpers/mock-data'
import PokemonList from './';
import PokemonItems from '../PokemonItems';

describe('PokemonList', () => {
  const mockProps = {
    data,
    isFetching: false,
    next: 'mock-url',
    showDetails: jest.fn(),
  }
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<PokemonList {...mockProps} />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all PokemonItems', () => {
    expect(wrapper.find(PokemonItems)).toHaveLength(data.length);
  });

  it('should render loader bar if data is fetching', () => {
    expect(wrapper.find('.list-footer')).toHaveLength(0);
    wrapper.setProps({
      isFetching: true,
    });
    wrapper.update();
    expect(wrapper.find('.list-footer')).toHaveLength(1);
    expect(wrapper.find('.list-footer').first().text()).toEqual('Fetching more data...');
  });

  it('should render end of the list message when next is null', () => {
    expect(wrapper.find('.list-footer')).toHaveLength(0);
    wrapper.setProps({
      next: null,
    });
    wrapper.update();
    expect(wrapper.find('.list-footer')).toHaveLength(1);
    expect(wrapper.find('.list-footer').first().text()).toEqual(`That's end of the list!`);
  });
});