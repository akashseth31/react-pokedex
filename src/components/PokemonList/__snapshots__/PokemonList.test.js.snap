// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList renders correctly 1`] = `ReactWrapper {}`;
